### PR TITLE
Log poller DB optimizations

### PR DESCRIPTION
**Pull request by connorwstein**
- Hex encoding not needed, postgres supports bytea comparisons, this doesn't appear to improve BenchmarkLogs or TestPopulateLoadedDB query time, but it does simplify the code.
- logs_idx_evm_id_event_address_block and  logs_idx_block_number indexes are not needed. The logs_idx covers them. No query speed difference in TestPopulateLoadedDB with them removed.
- Add typing for eventSig
- Fix query printing in debug sql

### CODE REVIEW

File: core/store/migrate/migrations/0117_add_log_poller_idx.sql [hunk 1]
```diff
@@ -3,5 +3,5 @@ CREATE INDEX logs_idx_block_number ON logs using brin(block_number);
 CREATE INDEX logs_idx_evm_id_event_address_block ON logs using btree (evm_chain_id,event_sig,address,block_number);
 
 -- +goose Down
-DROP INDEX logs_idx_block_number;
-DROP INDEX logs_idx_evm_id_event_address_block;
+DROP INDEX IF EXISTS logs_idx_block_number;
+DROP INDEX IF EXISTS logs_idx_evm_id_event_address_block;
```
Comment (jmank88) [inline core/store/migrate/migrations/0117_add_log_poller_idx.sql:7]: Is this valid even though we have already released these migrations? (I guess we don't hash them or anything?)

File: core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go [hunk 2]
```diff
@@ -612,7 +612,7 @@ func (c *coordinator) unmarshalLogs(
 	for _, lg := range logs {
 		rawLog := toGethLog(lg)
 		switch {
-		case bytes.Equal(lg.EventSig, c.randomnessRequestedTopic[:]):
+		case lg.EventSig == c.randomnessRequestedTopic:
```
Comment (jmank88) [inline core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go:615]: nit/ is this form an option now?
```suggestion
		switch lg.EventSig {
		case c.randomnessRequestedTopic:
```

File: core/store/migrate/migrations/0117_add_log_poller_idx.sql (hunk 1 above)
Comment (connorwstein) [reply core/store/migrate/migrations/0117_add_log_poller_idx.sql:7]: yeah all I see from goose is a single integer version corresponding to the migration number

File: core/services/pg/q.go [hunk 3]
```diff
@@ -280,10 +282,28 @@ func sprintQ(query string, args []interface{}) string {
...
+			pairs = append(pairs, fmt.Sprintf("$%d", i+1), fmt.Sprintf("'\\x%x'", v))
+		case common.Address:
+			pairs = append(pairs, fmt.Sprintf("$%d", i+1), fmt.Sprintf("'\\x%x'", v.Bytes()))
+		case common.Hash:
+			pairs = append(pairs, fmt.Sprintf("$%d", i+1), fmt.Sprintf("'\\x%x'", v.Bytes()))
+		case pq.ByteaArray:
+			s := fmt.Sprintf("('\\x%x'", v[0])
+			for j := 1; j < len(v); j++ {
+				s += fmt.Sprintf(",'\\x%x'", v[j])
+			}
```
Comment (jmank88) [inline core/services/pg/q.go:298]: Worth a [`strings.Builder`](https://pkg.go.dev/strings#Builder) to minimize allocs?
```suggestion
			var s strings.Builder
			fmt.Fprintf(&s, "('\\x%x'", v[0])
			for j := 1; j < len(v); j++ {
				fmt.Fprintf(&s, ",'\\x%x'", v[j])
			}
```

Comment (jmank88) [reply core/services/pg/q.go]: Might need to pass a `&s` or call `String()` below too :facepalm:

**Review by jmank88 — APPROVED** [review]

**Pull request merged**